core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -469,6 +456,41 @@ class CityConstructions {
                     builtBuildingUniqueMap.addUnique(unique)
     }
 
+    fun addFreeBuildings() {
+        // "Gain a free [buildingName] [cityFilter]"
+        val freeBuildingUniques = cityInfo.getLocalMatchingUniques(UniqueType.GainFreeBuildings, StateForConditionals(cityInfo.civInfo, cityInfo))
+
+        for (unique in freeBuildingUniques) {
+            val freeBuildingName = cityInfo.civInfo.getEquivalentBuilding(unique.params[0]).name
+            val citiesThatApply = when (unique.params[1]) {
```

Comment:
When { unique.isLocal, unique.isAntiLocal

Reply:
Not my code, and ... <sub><sup>(wtf is that isLocalEffect/isAntiLocalEffect anyway)</sub></sup> ... True, equivalent _and_ more efficient. OK, now I learned about those new (to me) helpers, I can include that.